clang-tools-extra/clangd/refactor/tweaks/ExtractVariable.cpp

```diff
@@ -176,7 +176,8 @@ bool ExtractionContext::exprIsValidOutside(const clang::Stmt *Scope) const {
   SourceLocation ScopeBegin = Scope->getBeginLoc();
   SourceLocation ScopeEnd = Scope->getEndLoc();
   for (const Decl *ReferencedDecl : ReferencedDecls) {
-    if (SM.isPointWithin(ReferencedDecl->getBeginLoc(), ScopeBegin, ScopeEnd) &&
+    if (ReferencedDecl->getBeginLoc().isValid() &&
```

Comment:
(I wondered whether there might be other cases where a `Decl` has an invalid location where we'd want this function to return `false`, but I couldn't think of a specific case, so I think this is fine for the time being.)